Content.Server/StationEvents/Metric/ChaosMetrics.cs

```diff
@@ -0,0 +1,241 @@
+using System.Linq;
+using System.Text.Json.Serialization;
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.chaos
+{
+    /// <summary>
+    ///     This class represents a collection of chaos
+    /// </summary>
+    /// <remarks>
+    ///     This is similar to the DamageSpecifier class.
+    /// </remarks>
+    [DataDefinition]
+    public sealed class ChaosMetrics : IEquatable<ChaosMetrics>
+    {
+        /// <summary>
+        ///     Main chaos dictionary. Most ChaosMetrics functions exist to somehow modifying this.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public Dictionary<string, FixedPoint2> ChaosDict { get; set; } = new();
```

Comment:
No hungarian notation variable names.

Also, why isn't this using a `record`, it seems like that's what you'd want as most operations already don't mutate it.

Reply:
This class is a clone of `DamageSpecifier` and has similar design choices about variable type, names etc.

There's going to be some maths and modification on chaos values later on, its just that the metrics stand alone and so don't need to do any maths on it. Most of those calculations will be going in the dynamic system.

Reply:
> This class is a clone of `DamageSpecifier` and has similar design choices about variable type, names etc.
> 
> There's going to be some maths and modification on chaos values later on, its just that the metrics stand alone and so don't need to do any maths on it. Most of those calculations will be going in the dynamic system.
> 
> I tried lifting the shared functionality into a generic base class but it got too hard around where the class was getting cloned.

I don't think there's any value in making a generic base class for this anyways. Unnecessary abstraction.